File: java_implementation/src/main/java/img/processing/Delete.java
package img.processing;
import java.awt.image.BufferedImage;

public class Delete {
    public static BufferedImage deleteArea(BufferedImage im, int width, int height, int x, int y, int areaWidth, int areaHeight,
            int channels) {
        // Checking if area is possible to delete
        BufferedImage deleted = im;
        if (x < 0 || y < 0 || x + areaWidth > width || y + areaHeight > height) {
            System.err.println("The area extends beyond the image size.");
            return null;
        }
        int whitePixel = (255 << 24) | (255 << 16) | (255 << 8) | 255;
        

        // Deleting area
        for (int i = y; i < y + areaHeight; i++) {
            for (int j = x; j < x + areaWidth; j++) {
                for (int c = 0; c < channels; c++) {
                    deleted.setRGB(x + j, y + i, whitePixel);
                }
            }
        }
        

        return deleted;
    }
}

File: python_implementation/delete.py
from PIL import Image

def delete_area(im, x, y, area_width, area_height, channels = 3):
    width, height = im.img.size
    if x < 0 or y < 0 or x + area_width > width or y + area_height > height:
        print("The area extends beyond the image size.")
        return None
    
    deleted = im.img.copy()

    # Deleting area
    white_pixel = (255, 255, 255, 255)
    for i in range(y, y + area_height):
        for j in range(x, x + area_width):
            for c in range(channels):
                deleted.putpixel((j, i), white_pixel)

    return deleted